source/dbo/Views/Power BI Reporting/vwUnmatchedDuplicates.sql

```diff
@@ -0,0 +1,82 @@
+﻿CREATE VIEW [dbo].[vwUnmatchedDuplicates]
+AS
+
+-- NTBS Notifications, with all notifications migrated in from ETS
+WITH NtbsNotifications AS (
+	SELECT n.NotificationId AS NtbsId,
+		CAST(n.ETSID AS INT) AS EtsId,
+		n.GroupId AS NtbsGroupId,
+		etsGroupings.GroupId AS EtsGroupId,
+		n.NotificationDate,
+		n.CreationDate,
+		p.Dob,
+		p.FamilyName,
+		REPLACE(p.NhsNumber, ' ', '') AS NhsNumber,
+		u.Username AS CaseManager
+	FROM [$(NTBS)].dbo.[Notification] n
+		INNER JOIN [$(NTBS)].dbo.Patients p ON n.NotificationId = p.NotificationId
+		INNER JOIN [$(NTBS)].dbo.HospitalDetails h ON n.NotificationId = h.NotificationId
+		LEFT JOIN [$(NTBS)].dbo.[User] u ON h.CaseManagerId = u.Id
+		LEFT JOIN [$(migration)].dbo.EtsGroupings etsGroupings ON n.ETSID = etsGroupings.EtsId
+	WHERE n.NotificationStatus IN ('Notified', 'Closed')
+),
+-- ETS notifications that have not been migrated into NTBS
+EtsNotifications AS (
+	SELECT NULL AS NtbsId,
+		n.LegacyId AS EtsId,
+		NULL AS NtbsGroupId,
+		etsGroupings.GroupId AS EtsGroupId,
+		n.NotificationDate,
+		n.AuditCreate AS CreationDate,
+		p.DateOfBirth,
+		p.Surname AS FamilyName,
+		REPLACE(p.NhsNumber, ' ', '') AS NhsNumber,
+		su.Username AS CaseManager
+	FROM [$(ETS)].dbo.[Notification] n
+		INNER JOIN [$(ETS)].dbo.Patient p ON n.PatientId = p.Id
+		INNER JOIN [$(ETS)].dbo.SystemUser su ON n.OwnerUserId = su.Id
+		LEFT JOIN [$(NTBS)].dbo.[Notification] ntbsNotification ON n.LegacyId = ntbsNotification.ETSID
+		LEFT JOIN [$(migration)].dbo.EtsGroupings etsGroupings ON n.LegacyId = etsGroupings.EtsId
+	WHERE n.AuditDelete IS NULL AND n.Submitted = 1 AND n.DenotificationId IS NULL
+		AND ntbsNotification.NotificationId IS NULL -- ie no corresponding NTBS migrated notification
+),
+AllNotifications AS (
+	SELECT *
+	FROM EtsNotifications
+
+	UNION
+
+	SELECT *
+	FROM NtbsNotifications
+)
+SELECT an2.DateOfBirth AS DateOfBirth,
+	an1.NhsNumber AS NhsNumber,
+	an1.NtbsId AS NtbsId1,
+	an1.ETSID AS EtsId1,
+	an1.NtbsGroupId AS NtbsGroupId1,
+	an1.NtbsGroupId AS EtsGroupId1,
+	an1.NotificationDate AS NotificationDate1,
+	an1.CreationDate AS CreationDate1,
+	an1.FamilyName AS FamilyName1,
+	an1.CaseManager AS CaseManager1,
+	an2.NtbsId AS NtbsId2,
+	an2.ETSID AS EtsId2,
+	an2.NtbsGroupId AS NtbsGroupId2,
+	an2.NtbsGroupId AS EtsGroupId2,
+	an2.NotificationDate AS NotificationDate2,
+	an2.CreationDate AS CreationDate2,
+	an2.FamilyName AS FamilyName2,
+	an2.CaseManager AS CaseManager2
+FROM AllNotifications an1
+	INNER JOIN AllNotifications an2
+		-- Notifications duplicate NHS numbers and dates of birth
+		ON an1.DateOfBirth = an2.DateOfBirth
+		AND an1.NhsNumber = an2.NhsNumber
+		-- But not the same notification twice
+		AND (an1.NtbsId IS NULL OR an2.NtbsId IS NULL OR an1.NtbsId <> an2.NtbsId)
```

Comment:
I don't think this accounts for it being the same notification but in its pre- and post-migration state. i.e. we don't want to discover that ETS 123 has the same NHS# and date of birth as NTBS 567 and it's because 567 was migrated in from 123.

Reply:
I think this is handled by the fact that migrated notifications are already filtered out of the `EtsNotifications`